src/main/scala/lms/core/stub.scala

```diff
@@ -748,6 +803,46 @@ trait LiftPrimitives {
  */
 trait PrimitiveOps extends Base with OverloadHack {
 
+  // def INT(x: Backend.Exp) = {
+  //   Adapter.typeMap(x) = manifest[Int]
+  //   new INT(x)
+  // }
+  case class INT(override val x: Backend.Exp)(implicit __pos: SourceContext) extends NUM(x) {
+    with_(__pos, manifest[Int])
+    // def +(y: INT): INT = INT(Adapter.g.reflect("+", x, y.x))
+    // def -(y: INT): INT = INT(Adapter.g.reflect("-", x, y.x))
+    // def *(y: INT): INT = INT(Adapter.g.reflect("*", x, y.x))
+    // def /(y: INT): INT = INT(Adapter.g.reflect("/", x, y.x))
+  }
+  def INT(i: Int)(implicit __pos: SourceContext): INT = INT(Backend.Const(i))
+
+  def NUM(x: Backend.Exp, m: Manifest[_])(implicit __pos: SourceContext): NUM = {
```

Comment:
```suggestion
  def NUM[T: Manifest](x: Backend.Exp)(implicit __pos: SourceContext): NUM = {
```
and `with_` can take `pos` and `m` as implicit parameters.

Reply:
The all CAPS classes and def constructors are supposed to be typeless (they don't take type parameters), so that the transformer function can use them. 
the `NUM[T: manifest]` would defeat the purpose.